fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushPredicateThroughJoin.java

```diff
@@ -122,15 +121,14 @@ private Plan pushDownPredicate(LogicalJoin<GroupPlan, GroupPlan> joinPlan,
         Expression left = ExpressionUtils.and(leftPredicates);
         Expression right = ExpressionUtils.and(rightPredicates);
         //todo expr should optimize again using expr rewrite
-        ExpressionRuleExecutor exprRewriter = new ExpressionRuleExecutor();
         Plan leftPlan = joinPlan.left();
         Plan rightPlan = joinPlan.right();
         if (!left.equals(BooleanLiteral.TRUE)) {
-            leftPlan = new LogicalFilter(exprRewriter.rewrite(left), leftPlan);
+            leftPlan = new LogicalFilter(left, leftPlan);
```

Comment:
the title is "normalize expression before rewriting", but here we remove normalizing stop before rewriting. Could you explain the reason for this change, please?

Reply:
The original `ExpressionRuleExecutor` mixes two types of rules, ones are used for normalization and the others for optimization. In pushing predicates stage, we don't need to do normalization. This pull request aims at separating these types of rules. If you think we can apply rules to optimize expressions, you can add code to do it.